### DISCUSSION TITLE
How to use onedrive client only for upload?

### DISCUSSION THREAD

**Discussion by ukreddy-erwin** [abraunegg/onedrive#1523] [Q&A] [answered]
I am using the below command to sync files to onedrive.

There is a folder in onedrive "Test_folder", so whichever files need to upload are being copied to the local folder with same name and once the sync command is ran, the files are being uploaded.

onedrive --synchronize --single-directory "Test_folder" --verbose

But, the files exists locally also and if we remove locally after syncing, next time, it is deleting online also.

Is there any version of this command to use only to upload to remote folder and no need to keep a local copy after upload?

**Comment by abraunegg** [comment, marked as answer]
@ukreddy-erwin 
> Is there any version of this command to use only to upload to remote folder and no need to keep a local copy after upload?

This is detailed in the usage document - please read: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-one-way-upload-sync

**Reply by ukreddy-erwin** [reply]
This is not working, can you please suggest:
onedrive --synchronize --sync-shared-folders --upload-only --no-remote-delete --single-directory 'Jenkins' --verbose

**Reply by ukreddy-erwin** [reply]
Even with below option, I am getting error. Folders are not syncing

`onedrive --synchronize --sync-shared-folders --single-directory 'Jenkins'`

```

Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
WARNING: The requested path for --single-directory does not exist locally. Creating requested path within /home/ubuntu/OneDrive
Syncing changes from selected OneDrive path ...
Uploading differences of Jenkins
Uploading new items of Jenkins
Retry sync count: 1: Key not found: fileSystemInfo
Syncing changes from selected OneDrive path ...
Uploading differences of Jenkins
Uploading new items of Jenkins
Retry sync count: 2: Key not found: fileSystemInfo
Syncing changes from selected OneDrive path ...
Uploading differences of Jenkins
Uploading new items of Jenkins
Giving up on sync after three attempts: Key not found: fileSystemInfo
std.json.JSONException@std/json.d(641): Key not found: fileSystemInfo
----------------
??:? _D3std9exception__T7bailOutHTCQBc4json13JSONExceptionZQBjFNaNfAyamMAxaZNn [0x55de732e0f93]
??:? pure @safe inout(std.json.JSONValue)* std.exception.enforce!(std.json.JSONException).enforce!(inout(std.json.JSONValue)*).enforce(inout(std.json.JSONValue)*, lazy const(char)[], immutable(char)[], ulong) [0x55de733a88f8]
??:? inout pure ref @safe inout(std.json.JSONValue) std.json.JSONValue.opIndex(immutable(char)[]) [0x55de733546fd]
??:? itemdb.Item sync.makeItem(ref const(std.json.JSONValue)) [0x55de73312d3e]
??:? void sync.SyncEngine.saveItem(std.json.JSONValue) [0x55de73327177]
??:? void sync.SyncEngine.uploadCreateDir(const(immutable(char)[])) [0x55de7332316f]
??:? void sync.SyncEngine.uploadNewItems(const(immutable(char)[])) [0x55de73321f36]
??:? void sync.SyncEngine.scanForDifferences(const(immutable(char)[])) [0x55de7331e3ba]
??:? void main.performSync(sync.SyncEngine, immutable(char)[], bool, bool, bool, long, bool, bool, bool, bool, monitor.Monitor) [0x55de73302449]
??:? _Dmain [0x55de73300ce0]
std.net.curl.CurlException@std/net/curl.d(4402): A libcurl function was given a bad argument on handle null
----------------
??:? _D3std9exception__T7bailOutHTCQBc3net4curl13CurlExceptionZQBnFNaNfAyamMAxaZNn [0x55de733a8cfe]
??:? pure @safe bool std.exception.enforce!(std.net.curl.CurlException).enforce!(bool).enforce(bool, lazy const(char)[], immutable(char)[], ulong) [0x55de733a8c6b]
??:? void std.net.curl.Curl._check(int) [0x55de73375c97]
??:? void std.net.curl.Curl.clear(etc.c.curl.CurlOption) [0x55de73376046]
??:? void std.net.curl.HTTP.clearRequestHeaders() [0x55de73374d9b]
??:? void onedrive.OneDriveApi.shutdown() [0x55de733064f4]
??:? _Dmain [0x55de73301ae6]
```

**Comment by abraunegg** [comment]
@ukreddy-erwin 
A few things here:

1. What application version are you running? Output of 'onedrive --version' is needed

This is required as you have an application crash, that 100% sounds like a prior bug that was fixed.

If you are running any application version not 2.4.12 or greater then you must upgrade your client to resolve the application crash.

If you are running the latest client version, you must open a new bug following the support process.

**Comment by ukreddy-erwin** [comment]
It is the latest version, I just compiled the repo clone before raising this issue on an aws EC2 ubuntu 20 machine

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by abraunegg** [comment]
@ukreddy-erwin 
> It is the latest version, I just compiled the repo clone before raising this issue on an aws EC2 ubuntu 20 machine

OK .. but *what* does the output of 'onedrive --version' say ..... 

Please follow what is being asked of you.

**Reply by ukreddy-erwin** [reply]
Last login: Tue Jun 29 20:13:51 2021 from 175.101.107.213
ubuntu@ip-172-31-15-89:~$ onedrive --version
onedrive v2.4.12-11-g65d289d

**Comment by abraunegg** [comment]
@ukreddy-erwin 
Please open a new bug report following the support process

**Comment by abraunegg** [comment]
@ukreddy-erwin
Please open a new bug report following the support process ... you MUST follow the correct support process and use the issue template to raise the issue, filling in *all* the details

**Reply by ukreddy-erwin** [reply]
I added with all the details that are relevant to the issue, please do the needful
